### PR TITLE
docs: fix some spelling

### PR DESCRIPTION
**Pull request by jsoref**
Fixes misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/liquidjs/actions/runs/9538354838#summary-26287456265

The action will report that the changes in this PR would make it mostly happy: https://github.com/jsoref/liquidjs/actions/runs/9538354866#summary-26287456395

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1113,7 +1113,7 @@ supports Node.js 6, other bundles are also provided via dist/liquid.esm.js, dist
 
 ### Features
 
-* at_least, at_most, sort_naturual for [#132](https://github.com/harttle/liquidjs/issues/132) ([e6f5f1c](https://github.com/harttle/liquidjs/commit/e6f5f1c))
+* at_least, at_most, sort_natural for [#132](https://github.com/harttle/liquidjs/issues/132) ([e6f5f1c](https://github.com/harttle/liquidjs/commit/e6f5f1c))
```
Comment (jsoref) [inline CHANGELOG.md:1116]: Some projects don't like changing changelogs. I'm happy to drop these changes if that's the case, but, personally I prefer being able to rely on content in them...

File: README.md [hunk 2]
```diff
@@ -6,12 +6,12 @@
 [![DUB license](https://img.shields.io/dub/l/vibe-d.svg?style=flat-square)](https://github.com/harttle/liquidjs/blob/master/LICENSE)
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg?style=flat-square)](https://github.com/harttle/liquidjs)
 
-A simple, expressive and safe [Shopify][shopify/liquid] / Github Pages compatible template engine in pure JavaScript.
-**The purpose of this repo** is to provide a standard Liquid implementation for the JavaScript community so that [Jekyll sites](https://jekyllrb.com), [Github Pages](https://pages.github.com/) and [Shopify templates](https://themes.shopify.com/) can be ported to Node.js without pain.
+A simple, expressive and safe [Shopify][shopify/liquid] / GitHub Pages compatible template engine in pure JavaScript.
+**The purpose of this repo** is to provide a standard Liquid implementation for the JavaScript community so that [Jekyll sites](https://jekyllrb.com), [GitHub Pages](https://pages.github.com/) and [Shopify templates](https://themes.shopify.com/) can be ported to Node.js without pain.
```
Comment (jsoref) [inline README.md:10]: brand

File: README.md [hunk 3]
```diff
@@ -80,7 +80,7 @@ Special thanks to the following sponsors!
     <tr>
       <td align="center" valign="top" width="14.28%"><a href="https://www.opensense.com/"><img src="https://images.opencollective.com/opensense-inc/bf840ae/logo/256.png?height=100" width="100px;" alt="Opensense Inc."/><br /><sub><b>Opensense</b></sub></a></td>
       <td align="center" valign="top" width="14.28%"><a href="https://www.11ty.dev/"><img src="https://avatars.githubusercontent.com/u/35147177?v=4?s=100" width="100px;" alt="Eleventy"/><br /><sub><b>Eleventy</b></sub></a></td>
-      <td align="center" valign="top" width="14.28%"><a href="http://about.me/peterdehaan"><img src="https://avatars2.githubusercontent.com/u/557895?v=4?s=100" width="100px;" alt="Peter deHaan"/><br /><sub><b>Peter deHaan</b></sub></a></td>
+      <td align="center" valign="top" width="14.28%"><a href="https://about.me/peterdehaan"><img src="https://avatars2.githubusercontent.com/u/557895?v=4?s=100" width="100px;" alt="Peter deHaan"/><br /><sub><b>Peter deHaan</b></sub></a></td>
```
Comment (jsoref) [inline README.md:83]: I've generally changed `http:` to `https:` for cases where the page appears valid.

File: benchmark/engines/create-liquid.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ function createEngine (pkg) {
     cache: true,
     extname: '.liquid'
   })
-  liquid.registerFilter('url', path => `http://example.com${path}`)
+  liquid.registerFilter('url', path => `https://example.com${path}`)
```
Comment (jsoref) [inline benchmark/engines/create-liquid.js:10]: this includes `example.com` which is available via https (happy to drop `example.com` specifically if requested)

File: benchmark/tag.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ function tag () {
     new Benchmark.Suite('tag')
       .add('if', test('{% if "foobar" %}foo{% endif %}'))
       .add('unless', test('{%unless "foo"%}true{%else%}false{%endunless%}'))
-      .add('for', test('{% for i in (1..3) %}{{fooloop.index}}{% endfor %}'))
+      .add('for', test('{% for i in (1..3) %}{{forloop.index}}{% endfor %}'))
```
Comment (jsoref) [inline benchmark/tag.js:13]: I'm assuming this is supposed to correspond with `'for'` ...)

File: bin/character-gen.js [hunk 6]
```diff
@@ -10,7 +10,7 @@ const isInlineBlank = c => c === '\t' || c === ' ' || c === '\r'
 const isSign = c => c === '-' || c === '+'
 // See https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/RegExp
 const unicodeBlanks = '\u00a0\u1680\u180e\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u2028\u2029\u202f\u205f\u3000'
-const unicodePunctuations = '“”'
+const unicodePunctuation = '“”'
```
Comment (jsoref) [inline bin/character-gen.js:13]: [punctuation](https://www.dictionary.com/browse/punctuation) here (3) is [`punctuation marks`](https://www.dictionary.com/browse/punctuation-mark) -- it already refers to punctuation marks -- you wouldn't write "one punctuation" although you could write "one punctuation mark".

File: demo/reactjs/README.md [hunk 7]
```diff
@@ -2201,7 +2201,7 @@ or for a GitHub user page:
 or for a custom domain page:
 
 ```json
-  "homepage": "https://mywebsite.com",
+  "homepage": "http://mywebsite.com",
```
Comment (jsoref) [inline demo/reactjs/README.md:2204]: This is a very odd special case. `mywebsite.com` should really be `example.<something>` (com, org, net, -- there are a number of choices) -- the domain `mywebsite.com` is for sale and could be squatted on at any moment by a malicious entity. There are 6 instances of `mywebsite.com` in this file, and only one of them uses `https`. I'm happy to change them all to use `https`, but I'd be happier to change all of them not to be `mywebsite.com`.

File: docs/source/_data/sidebar.yml [hunk 8]
```diff
@@ -9,7 +9,7 @@ tutorials:
   advanced:
     caching: caching.html
     escaping: escaping.html
-    registeration: register-filters-tags.html
+    registration: register-filters-tags.html
```
Comment (jsoref) [inline docs/source/_data/sidebar.yml:12]: This could break things

File: docs/themes/navy/layout/partial/head.swig [hunk 9]
```diff
@@ -1,4 +1,4 @@
-<head prefix="og: http://ogp.me/ns#">
+<head prefix="og: https://ogp.me/ns#">
```
Comment (jsoref) [inline docs/themes/navy/layout/partial/head.swig:1]: This is recommended on https://ogp.me/ ...

File: docs/themes/navy/source/css/_variables.styl [hunk 10]
```diff
@@ -22,7 +22,7 @@ vendor-prefixes = webkit moz ms official
         --color-default-invert: #444
         --color-gray: invert(#999)
         --color-border: #6a6a6a
-        --color-backgrond: #fff
+        --color-background: #fff
```
Comment (jsoref) [inline docs/themes/navy/source/css/_variables.styl:25]: this may be a breaking change...

File: test/e2e/xhr.spec.ts [hunk 11]
```diff
@@ -113,10 +113,10 @@ describe('xhr', () => {
     })
     it('should support with url', async () => {
       engine = new LiquidUMD({
-        root: 'https://foo.com/bar/',
+        root: 'http://foo.com/bar/',
```
Comment (jsoref) [inline test/e2e/xhr.spec.ts:116]: This, like mywebsite.com, is another example where `example.com` or a friend should really be used. `foo.com` isn't owned by liquidjs and could be malicious at any time. There were 8 instances of `foo.com` and only 2 were using `https:`.

File: README.md (hunk 3 above)
Comment (harttle) [reply README.md:83]: this part is auto generated but I will merge this anyway

File: docs/themes/navy/source/css/_variables.styl (hunk 10 above)
Comment (harttle) [reply docs/themes/navy/source/css/_variables.styl:25]: not referenced :)

File: docs/source/_data/sidebar.yml (hunk 8 above)
Comment (harttle) [reply docs/source/_data/sidebar.yml:12]: fixed for zh_ch.yml

**Comment by harttle** [comment]
@all-contributors please add @jsoref for docs

**Pull request merged**